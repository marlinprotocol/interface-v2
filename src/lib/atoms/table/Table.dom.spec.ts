import { render } from '@testing-library/svelte';
import TableComponent from './Table.svelte';
import html from '@playpilot/svelte-htm';

describe('TableComponent', () => {
	const tableHeadingMock = [
		{ id: '1', title: 'Heading 1' },
		{ id: '2', title: 'Heading 2' },
		{ id: '3', title: 'Heading 3' }
	];
	const handleSortDataMock = vi.fn((id: string) => {
		console.log(id);
	});

	it("render's table correctly", () => {
		const { container } = render(TableComponent, {
			tableHeading: tableHeadingMock,
			styleClass: 'custom-table-style',
			headingStyleClass: 'custom-heading-style',
			handleSortData: handleSortDataMock,
			iconWidth: '16px'
		});
		expect(container).toMatchSnapshot();
	});
	it('renders table headings and applies styles correctly', () => {
		const { getByTestId, getByText } = render(TableComponent, {
			tableHeading: tableHeadingMock,
			styleClass: 'custom-table-style',
			headingStyleClass: 'custom-heading-style',
			handleSortData: handleSortDataMock,
			iconWidth: '16px'
		});

		// Check if the table container has the correct classes
		expect(getByTestId('table-container').className).contain('custom-table-style');

		// Check if each heading is rendered with a TableHeadingText component and has the correct classes
		for (const heading of tableHeadingMock) {
			expect(getByText(heading.title).innerHTML).toBe(heading.title);
			expect(getByText(heading.title).closest('th')?.innerHTML).contain('custom-heading-style');
		}
	});

	it('should render component inside slot (table body) properly', async () => {
		const { findByText } = render(html`
                <${TableComponent} tableHeading=${tableHeadingMock}>
                 <div slot='tableBody'>table-body</div>
                </${TableComponent}>
              `);

		expect(await findByText('table-body')).toBeTruthy();
	});

	// Additional tests can be written based on the need, such as checking the tooltip directions, etc.
});
