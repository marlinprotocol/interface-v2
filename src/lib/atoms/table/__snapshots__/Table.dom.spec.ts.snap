// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TableComponent > render's table correctly 1`] = `
<body>
  <div>
    <div
      class="custom-table-style"
      data-testid="table-container"
    >
      <table
        class="w-full rounded-[18px] text-center"
      >
        <thead
          class="w-full"
        >
          <tr>
            <th
              class="bg-[#F0F0F0] py-[26px] first:rounded-tl-[18px] last:rounded-tr-[18px]"
            >
              <div
                class="mt-0.5 flex flex-nowrap items-center justify-start gap-1 text-center text-[14px] font-normal text-[#030115] custom-heading-style ml-4"
                data-testid="table-heading-text"
              >
                <div
                  class="relative flex items-center gap-1"
                >
                  <span
                    class="w-fit tracking-widest"
                  >
                    Heading 1
                  </span>
                   
                </div>
              </div>
              <!--&lt;TableHeadingText&gt;-->
               
            </th>
            <th
              class="bg-[#F0F0F0] py-[26px] first:rounded-tl-[18px] last:rounded-tr-[18px]"
            >
              <div
                class="mt-0.5 flex flex-nowrap items-center justify-start gap-1 text-center text-[14px] font-normal text-[#030115] custom-heading-style"
                data-testid="table-heading-text"
              >
                <div
                  class="relative flex items-center gap-1"
                >
                  <span
                    class="w-fit tracking-widest"
                  >
                    Heading 2
                  </span>
                   
                </div>
              </div>
              <!--&lt;TableHeadingText&gt;-->
               
            </th>
            <th
              class="bg-[#F0F0F0] py-[26px] first:rounded-tl-[18px] last:rounded-tr-[18px]"
            >
              <div
                class="mt-0.5 flex flex-nowrap items-center justify-start gap-1 text-center text-[14px] font-normal text-[#030115] custom-heading-style"
                data-testid="table-heading-text"
              >
                <div
                  class="relative flex items-center gap-1"
                >
                  <span
                    class="w-fit tracking-widest"
                  >
                    Heading 3
                  </span>
                   
                </div>
              </div>
              <!--&lt;TableHeadingText&gt;-->
               
            </th>
          </tr>
        </thead>
         
      </table>
    </div>
    <!--&lt;Table&gt;-->
  </div>
</body>
`;
